webapp/src/lib/Ui.ts

```diff
@@ -24,42 +32,79 @@ export async function showSnack(snack: MatSnackBar, text: string, cmd: () => voi
     }
 }
 
+/**
+ * Shows a simple dialog with an OK and a Cancel button. The return value of the
+ * method is a boolean: true for OK and false for Cancel.
+ *
+ * @param dialog reference to the matDialog
+ * @param title shown in h1 in the dialog
+ * @param text the text in HTML
+ * @param buttons can override the text of the OKButton ("OK") or the CancelButton ("Cancel")
+ */
 export async function showDialogOKC(dialog: MatDialog, title: string, text: string,
-                                    result: (result: boolean) => void,
-                                    buttons: IDialogOKCButtons = {OKButton: "OK", CancelButton: "Cancel"}) {
+                                    buttons: IDialogOKCButtons = {OKButton: "OK", CancelButton: "Cancel"}):
+    Promise<boolean> {
     const tc = dialog.open(DialogOKCancelComponent, {data: {Title: title, Text: text, Buttons: buttons}});
-    tc.afterClosed().subscribe(result);
+    return new Promise((resolve) => tc.afterClosed().subscribe((result) => resolve(result)));
 }
 
+/**
+ * Shows a simple dialog with a dismiss button and returns when the user presse the dismiss button.
+ *
+ * @param dialog reference to the matDialog
+ * @param title shown in h1 in the dialog
+ * @param text the text in HTML
+ * @param dismiss the string to be shown in the dismiss button
+ */
 export async function showDialogInfo(dialog: MatDialog, title: string, text: string,
-                                     dismiss: string,
-                                     result: (result: boolean) => void = null) {
+                                     dismiss: string) {
     const tc = dialog.open(DialogOKCancelComponent, {
         data: {
             Buttons: {OKButton: dismiss, CancelButton: ""},
             Text: text,
             Title: title,
         },
     });
-    tc.afterClosed().subscribe((res) => {
-        if (result) {
-            result(res);
-        }
-    });
+    return new Promise((resolve) => tc.afterClosed().subscribe(() => resolve()));
 }
 
+/**
+ * Convenience method for showTransactions that only shows one progress step: "Storing Credential".
+ *
+ * @param dialog reference to the matDialog
+ * @param title shown in h1 in the dialog
+ * @param store the callback to the actual storing of the credential.
+ */
 export async function storeCredential(dialog: MatDialog, title: string,
-                                      store: () => Promise<void>) {
+                                      store: <T>() => Promise<T | void>) {
```

Comment:
`T | void` is weird to read (and should typecast to `T` anyway). As you're `return`ing the result of `store` also,  `storeCredential` should become `storeCredential<T>` (the type is not bound to `store` only in this case).

```suggestion
                                      store: () => Promise<T>): Promise<T> {
```

Reply:
`storeCredential` is a convenience function that only does one thing: storing the credential in the blockchain. So I replaced the callback with a parameter of type `Data`. While replacing all calls to `storeCredential`, I couldn't resist to refactor `ContactsComponent`, too...